tools/zulint/custom_rules.py

```diff
@@ -6,192 +6,233 @@
-                            " for interpreters other than sh."},
-            {'pattern': '^#!/usr/bin/env python$',
-             'description': "Use `#!/usr/bin/env python3` instead of `#!/usr/bin/env python`."}
-        ]  # type: RuleList
-        shebang_rules_to_apply = get_rules_applying_to_fn(fn=fn, rules=shebang_rules)
-        for rule in shebang_rules_to_apply:
-            if re.search(rule['pattern'], firstline):
-                print_err(identifier, color,
-                          '{} at {} line 1:'.format(rule['description'], fn))
-                print_err(identifier, color, firstline)
+        # TODO: Move the below into more of a framework.
+        firstline = None
+        lastLine = None
+        if line_tups:
+            # line_fully_stripped for the first line.
+            firstline = line_tups[0][3]
+            lastLine = line_tups[-1][1]
+
+        if max_length is not None:
+            ok = self.check_file_for_long_lines(
+                fn=fn,
+                max_length=max_length,
+                line_tups=line_tups,
+            )
+            if not ok:
                 failed = True
 
-    if lastLine and ('\n' not in lastLine):
-        print("No newline at the end of file.  Fix with `sed -i '$a\\' %s`" % (fn,))
-        failed = True
+        if firstline:
+            shebang_rules = [
+                        {'pattern': '^#!',
+                         'description': "zerver library code shouldn't have a shebang line.",
+                         'include_only': set(['zerver/'])},
+                        # /bin/sh and /usr/bin/env are the only two binaries
+                        # that NixOS provides at a fixed path (outside a
+                        # buildFHSUserEnv sandbox).
+                        {'pattern': '^#!(?! *(?:/usr/bin/env|/bin/sh)(?: |$))',
+                         'description': "Use `#!/usr/bin/env foo` instead of `#!/path/foo`"
+                                        " for interpreters other than sh."},
+                        {'pattern': '^#!/usr/bin/env python$',
+                         'description': "Use `#!/usr/bin/env python3` instead of `#!/usr/bin/env python`."}
+            ]  # type: Rule
+            shebang_rules_to_apply = self.get_rules_applying_to_fn(fn=fn, rules=shebang_rules)
+            for rule in shebang_rules_to_apply:
+                if re.search(rule['pattern'], firstline):
+                    print_err(identifier, color,
+                              '{} at {} line 1:'.format(rule['description'], fn))
+                    print_err(identifier, color, firstline)
+                    failed = True
+
+        if lastLine and ('\n' not in lastLine):
+            print("No newline at the end of file. Fix with `sed -i '$a\\' %s`" % (fn,))
+            failed = True
 
-    return failed
+        return failed
+
+    def check(self, by_lang):
+        # type: (Dict[str, List[str]]) -> bool
```

Comment:
@timabbott there were no real line wrap changes required except here which are just comments so I have left them in the same commit. (The line wrap changes you saw earlier were according to max_length of 79(from zt) which I did it here by mistake) I have moved them back to original.

Reply:
Yeah, line-wrapping of the comments I wanted in a separate commit just because they confuse `git show --color-moved --color-moved-ws=ignore-all-space` into thinking a bigger change happened